src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Unix.cs

```diff
@@ -41,6 +41,16 @@ public static bool IsProcessRunning(int processId)
             return CreateProcessInfo(processId);
         }
 
+        /// <summary>Gets the process name for the specified process ID on the specified machine.</summary>
+        /// <param name="processId">The process ID.</param>
+        /// <param name="machineName">The machine name.</param>
+        /// <returns>The process name for the process if it could be found; otherwise, null.</returns>
+        public static string? GetProcessName(int processId, string machineName)
```

Comment:
This now-unused method can be deleted. :)

Reply:
I tried that, but I get a compilation error:
`8>C:\git\runtime\src\libraries\System.Diagnostics.Process\src\System\Diagnostics\Process.cs(965,51,965,65): error CS0117: 'ProcessManager' does not contain a definition for 'GetProcessName'`

Reply:
Of course three comments were related, so this alone will not work.

Reply:
I took another look at this just now and I think that each `ProcessManager.*.cs` needs its own copy of this method since it is called from `EnsureState` in `Process.cs` (which I _think_ is for all platforms).
I could very well be wrong though as this is my first multi-targetted solution that I've worked on, so do feel free to jump in and make any changes!

Reply:
As pointed out in other comment, if you revert the changes made in EnsureState then this method is not required on Unix.